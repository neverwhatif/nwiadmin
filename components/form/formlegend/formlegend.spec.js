import React from 'react';
import { shallow } from 'enzyme';

import FormLegend from './index';

jest.mock('nwiadmin/components/flag', () => 'Flag');

describe('FormLegend', () => {
    const component = shallow(<FormLegend>Lorem</FormLegend>);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(component.text()).toBe('Lorem');
    });
});
