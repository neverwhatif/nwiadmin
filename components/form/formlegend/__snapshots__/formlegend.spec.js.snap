// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLegend should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormLegend
    isOffset={false}
  >
    Lorem
  </FormLegend>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Flag
        isOffset={false}
      >
        Lorem
      </Flag>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lorem",
        "isOffset": false,
      },
      "ref": null,
      "rendered": "Lorem",
      "type": "Flag",
    },
    "type": "legend",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Flag
          isOffset={false}
        >
          Lorem
        </Flag>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "isOffset": false,
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "Flag",
      },
      "type": "legend",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
